hw2/03-count.js

```diff
@@ -1,3 +1,34 @@
-// Add your code here
+// Stuart Rimel
 
-input.addEventListener('keydown', handleKeyDown);
+const inputElement = document.getElementById('search-input');
+const textElement = document.getElementById('text');
+const originalInnerHTML = textElement.innerHTML;
+
+const highlight = function highlight(word, searchText) {
+  return word.replace(
+    searchText,
+    `<span style="background-color: yellow">${searchText}</span>`
+  );
+};
+
+const findAndHighlight = function findAndHighlight(searchText) {
+  let words = originalInnerHTML.split(' ');
+  for (i = 0; i < words.length; i++) {
+    if (words[i].includes(searchText)) {
+      words[i] = highlight(words[i], searchText);
+    }
+  }
+  textElement.innerHTML = words.join(' ');
+};
+
+const handleKeyDown = function handleKeyDown(event) {
+  let searchText = event.target.value;
+  if (event.key.length === 1) {
+    searchText += event.key;
+  } else if (event.key === 'Backspace') {
+    searchText = searchText.slice(0, -1);
+  }
+  findAndHighlight(searchText);
+};
+
+inputElement.addEventListener('keydown', handleKeyDown);
```

Comment:
I've read some documentation about keydown and keyup, and I think if we use "keyup" event, we will not need to implement the logic from lines 26-30. This would reduce the complexity.

Reply:
Yeah I think I would of done it a different way, but doesn't the assignment want us to use "keydown"?

Reply:
I think the requirement is "Consider using string and array methods to complete this exercise." I think for this exercise, if we did some research about it, we can use what is applicable for the implementation.

Reply:
I see, thanks for the feedback! I agree that using keyup would be easier and result in less code. However, I think I'll leave it as "keydown" since the requirements aren't explicit, but we are provided with a keydown event listener in the starting file and the keydown documentation is linked as a resource for this exercise. I'd also argue that any added complexity by using "keydown" over "keyup" is negligible. If anything it's good practice for keydown events!